### ISSUE TITLE
[feature request] Make structured logging configurable

### ISSUE THREAD

**Issue by rossmcdonald** [influxdata/influxdb#7940]
### Feature Request

__Proposal:__ 

We should allow users to configure the format of the logs via configuration.

__Current behavior:__

Configuring the log formatting is not possible.

__Desired behavior:__

Configuring the log format should be possible, ie setting logs to be in JSON, logfmt, etc format.

__Use case:__ 

InfluxDB logs using a structured logging library, which allows for log messages to expose more colorful metrics about the underlying process in a machine-readable format. The current formatting is not great, and is difficult for both a human and machine to read accurately.

**Comment by jsternberg** [comment]
One potential problem with configurable logging are these three messages:

```
[I] 2017-02-06T16:31:57Z InfluxDB starting, version unknown, branch unknown, commit unknown
[I] 2017-02-06T16:31:57Z Go version go1.7.4, GOMAXPROCS set to 8
[I] 2017-02-06T16:31:57Z no configuration provided, using default settings
```

All three of these go through the log and should be output with the configured logging, but all of them happen before a log file is even read. What we could do is load the config earlier and delay these messages until after logging is configured. That would probably be the best way.

**Comment by Artanicus** [comment]
Also would love an option to be able to drop the timestamp. I'm feeding them into rsyslog anyway which adds it's own structured timestamp and makes log analysis feasible but further processing is complicated by similar messages being dis-similar due to the extra timestamp in the message field.

**Comment by jsternberg** [comment]
@Artanicus can you file a new issue for that? I'll ensure it is considered for the 1.6 cycle. We just got this into the 1.5 cycle and will not have time to do what you want.

This is implemented in #9056.
